### PR TITLE
Update DPDK version to 17.11

### PR DESCRIPTION
**Pull request by shinae-woo**
* libnuma becomes mandatory
* port_id becomes to 16 bits
* Extending supports for various RX offloads and associated API changes
* A few minior API changes

### CODE REVIEW

File: core/Makefile [hunk 1]
```diff
@@ -83,7 +83,7 @@ endif
 # user requests a static build.
 ALWAYS_DYN_LIBS := -lpthread -ldl
 # These libraries are not supported by pkg-config.
-ALWAYS_LIBS := -lpcap -lgflags
+ALWAYS_LIBS := -lpcap -lgflags -lnuma
 # If pkg-config is available, we just need a list of the dependecies.
```
Comment (shinae-woo) [inline core/Makefile]: @ddiproietto  Could you confirm this change are appropriate?

**Comment by shinae-woo** [comment]
The container should be re-build. ❓ ❓

File: core/Makefile (hunk 1 above)
Comment (ddiproietto) [reply core/Makefile]: The change makes sense, thanks

**Comment by shinae-woo** [comment]
@sangjinhan Thank you for building new containers.

**Comment by shinae-woo** [comment]
@ddiproietto Thank you for the review! I will fix that point!

**Comment by sangjinhan** [comment]
In `core/modules/ip_lookup.cc`, there is some temporary code to work around an issue of DPDK 17.02. Given that the issue has gone in the new version, could you clean up the hack I made in the file?

**Comment by shinae-woo** [comment]
@sangjinhan This is a very weird bug on g++ side... Anyway, DPDK 17.11 have the similar workaround codes inside `rte_lpm_lookupx4`, so I remove the temporary code from the module code.

**Comment by shinae-woo** [comment]
image

**Comment by sangjinhan** [comment]
That's the code I pushed to the DPDK upstream 😸

**Comment by sangjinhan** [comment]
Merged, thanks!
tumblr_m83dqfx91m1qzex9io1_1280

**Pull request merged**